src/valid.rs

```diff
@@ -50,15 +50,12 @@ pub trait Validator<A, E, T>: Sized {
         Fusion(self.zip(other))
     }
 
-    fn trace(self, trace: T) -> Valid<A, E, T>
-    where
-        T: Clone,
-    {
+    fn trace(self, trace: impl Into<T> + Clone) -> Valid<A, E, T> {
```

Comment:
```suggestion
    fn trace(self, trace: impl Into<T>) -> Valid<A, E, T> {
```
We shouldn't put a hard constraint on the type T to implement Clone.

Reply:
Nevermind.